### PR TITLE
CAPBM: migrate consumerRef to BareMetalMachine

### PR DESCRIPTION
**Pull request by Jaakko-Os**

### CODE REVIEW

File: baremetal/machine_manager_test.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func TestChooseHost(t *testing.T) {
 			ConsumerRef: &corev1.ObjectReference{
 				Name:       "someothermachine",
 				Namespace:  "myns",
-				Kind:       "Machine",
+				Kind:       "BMMachine",
 				APIVersion: capi.GroupVersion.String(),
```
Comment (maelk) [inline baremetal/machine_manager_test.go]: capbm ?

File: baremetal/machine_manager_test.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func TestChooseHost(t *testing.T) {
 			ConsumerRef: &corev1.ObjectReference{
 				Name:       "machine1",
 				Namespace:  "myns",
-				Kind:       "Machine",
+				Kind:       "BMMachine",
 				APIVersion: capi.GroupVersion.String(),
```
Comment (maelk) [inline baremetal/machine_manager_test.go]: capbm ?

File: baremetal/machine_manager_test.go (hunk 1 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager_test.go]: done

File: baremetal/machine_manager_test.go (hunk 2 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager_test.go]: done

**Pull request merged**